src/pytorch_lightning/tuner/batch_size_scaling.py

```diff
@@ -69,94 +53,120 @@ def scale_batch_size(
     if trainer.progress_bar_callback:
         trainer.progress_bar_callback.disable()
 
-    # Initially we just double in size until an OOM is encountered
-    new_size, _ = _adjust_batch_size(trainer, batch_arg_name, value=init_val)  # initially set to init_val
+    new_size, _ = _adjust_batch_size(trainer, batch_arg_name, value=init_val)
+
     if mode == "power":
-        new_size = _run_power_scaling(trainer, model, new_size, batch_arg_name, max_trials)
+        new_size = _run_power_scaling(trainer, model, new_size, batch_arg_name, max_trials, params)
     elif mode == "binsearch":
-        new_size = _run_binsearch_scaling(trainer, model, new_size, batch_arg_name, max_trials)
-    else:
-        raise ValueError("mode in method `scale_batch_size` could either be `power` or `binsearch`")
+        new_size = _run_binary_scaling(trainer, model, new_size, batch_arg_name, max_trials, params)
+
+    _collect_garbage()
 
-    garbage_collection_cuda()
     log.info(f"Finished batch size finder, will continue with full run using batch size {new_size}")
 
     __scale_batch_restore_params(trainer, params)
 
     if trainer.progress_bar_callback:
         trainer.progress_bar_callback.enable()
 
-    # Restore initial state of model
     trainer._checkpoint_connector.restore(ckpt_path)
     trainer.strategy.remove_checkpoint(ckpt_path)
 
     return new_size
 
 
 def __scale_batch_dump_params(trainer: "pl.Trainer") -> Dict[str, Any]:
-    return {
-        "max_steps": trainer.fit_loop.max_steps,
-        "logger": trainer.logger,
+    dumped_params = {
+        "loggers": trainer.loggers,
         "callbacks": trainer.callbacks,
-        "auto_scale_batch_size": trainer.auto_scale_batch_size,
-        "auto_lr_find": trainer.auto_lr_find,
-        "limit_train_batches": trainer.limit_train_batches,
     }
+    if trainer.state.fn == "fit":
+        loop = trainer.fit_loop
+        dumped_params["max_steps"] = trainer.max_steps
+        dumped_params["limit_val_batches"] = trainer.limit_val_batches
```

Comment:
Isn't limit_train_batches missing now? It was there in the previous code

Reply:
added back!